DataLink.tex

```diff
@@ -888,26 +887,50 @@ \subsection{Service Resources}
 parse the URL to decide how to append additional parameters when
 invoking the service.
 
-In case the contentType is ``text/html'' the client SHOULD send the result 
+In case the contentType is ``text/html'', the client SHOULD send the result 
 of the service query to a web browser.  This is appropriate for both HTML
-documents or web interactive interface. 
+documents and web interactive interfaces. 
  
-A service descriptor may contain more than a single exampleURL.
+A service descriptor may contain multiple exampleURL PARAMs.
 In exampleURL PARAMs, operators can give valid service calls as GET-able
-URLs in the PARAMs' value attributes. They are intended as an aid for
+URLs in the PARAMs' value attribute. They are intended as an aid for
 debugging, in particular to aid users and developers in making sure a
-service is still operating as expected. The PARAM's description should
+service is still operating as expected. The PARAM description should
 give an indication of what the call will result in. End-user clients
-might indicate exampleURLs to the user after service failures.
-
-A GROUP with \attval{name}{inputParams} contains PARAM elements describing
-how to invoke the service. For services where the parameter values
-come from columns in the results resource, we use the ref attribute of
-the PARAM to indicate the FIELD (column) with the values. Other PARAM
-elements (without a ref attribute) are also allowed; these would describe
-additional service parameters, the type of value that must be specified,
-the meaning (UCD) of the value they apply to, etc.
-
+might indicate exampleURLs to the user after unexpected service failures.
+
+\subsection{Input PARAMs}
+
+A service descriptor must contain a GROUP element with \attval{name}{inputParams} 
+to describe user-specified input parameters of the service. There are three types of 
+input params: params with a fixed value, params where the values come from the 
+``results'', and params where the value is variable and chosen/specified by the user.
+
+For params with a fixed value (e.g. \attval{fly}{true}), the client {\bf must}
+treat it as a required parameter and include it in the service invocation; this allows
```

Comment:
It's not quite how I interpreted it up till now; the supplied value could be seen/intended as a default/suggested value which clients could allow users to see and change. If this text is actually what's intended I can change my client behaviour. Or, this MUST could be downgraded to SHOULD.  Similar comment applies to referenced values.